### ISSUE TITLE
heddle() should replicate but not replace when given unquoted NAs

### ISSUE THREAD

**Issue by mikemahoney218** [mikemahoney218/heddlr#20]
Right now, you can replicate patterns for each row in a dataset by using `heddle` with a placeholder phrase that isn't matched in the pattern anywhere. However, if it turns out that the string is in the pattern text, it gets replaced -- even if the string in question is an unquoted `NA`. This is problematic for two reasons:

1. If you provide an unquoted NA as your string to be replaced, you're clearly signalling you want nothing replaced
2. If you provide something _else_ which doesn't match, that should raise a warning -- we shouldn't encourage a risky behavior as the default way to replicate without replacement

I should implement changes for both of those behaviors, so that the first example of both vector and dataframe implementations below raises a warning (maybe an error?) and the second returns the first. 

```` r
library(heddlr)
x <- "## True Facts\n\n```{r}\nprint(\"heddlr is great!\")\n```\n\n"
spList <- unique(iris$Species)
heddle(spList, x, NA)
#> [1] "## True Facts\n\n```{r}\nprint(\"heddlr is great!\")\n```\n\n"
#> [2] "## True Facts\n\n```{r}\nprint(\"heddlr is great!\")\n```\n\n"
#> [3] "## True Facts\n\n```{r}\nprint(\"heddlr is great!\")\n```\n\n"
heddle(spList, "NA NA NA NA", NA)
#> [1] "setosa setosa setosa setosa"                
#> [2] "versicolor versicolor versicolor versicolor"
#> [3] "virginica virginica virginica virginica"

library(dplyr) 
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
iris %>%
  distinct(Species) %>%
  mutate(template = heddle(Species, x, NA))
#>      Species                                                    template
#> 1     setosa ## True Facts\n\n```{r}\nprint("heddlr is great!")\n```\n\n
#> 2 versicolor ## True Facts\n\n```{r}\nprint("heddlr is great!")\n```\n\n
#> 3  virginica ## True Facts\n\n```{r}\nprint("heddlr is great!")\n```\n\n
iris %>%
  distinct(Species) %>%
  mutate(template = heddle(Species, "NA NA NA NA", NA))
#>      Species                                    template
#> 1     setosa                 setosa setosa setosa setosa
#> 2 versicolor versicolor versicolor versicolor versicolor
#> 3  virginica     virginica virginica virginica virginica
````

<sup>Created on 2020-01-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mikemahoney218** [comment]
Now that I'm actually coding this, the first example of each case shouldn't give a warning, since it's using an unquoted NA :sweat_smile:

**Comment by mikemahoney218** [comment]
Should be fixed in 100d436
